biodatasets/test_psytar/test_psytar.py

```diff
@@ -0,0 +1,482 @@
+               Hadi Kharrazi and
+               Anthony Faiola and
+               Yi Shuan Shirley Wu and
+               Christina E. Eldredge and
+               Jake Luo and
+               Mike Conway and
+               Jiaxi Zhu and
+               Soo Kyung Park and
+               Kelly Xu and
+               Hamideh Moayyed and
+               Somaieh Goudarzvand},
+  title     = {A systematic approach for developing a corpus of patient \
+               reported adverse drug events: A case study for {SSRI} and {SNRI} medications},
+  journal   = {Journal of Biomedical Informatics},
+  volume    = {90},
+  year      = {2019},
+  url       = {https://doi.org/10.1016/j.jbi.2018.12.005},
+  doi       = {10.1016/j.jbi.2018.12.005},
+}
+"""
+
+_DATASETNAME = "psytar"
+
+_DESCRIPTION = """\
+The "Psychiatric Treatment Adverse Reactions" (PsyTAR) dataset contains 891 drugs
+reviews posted by patients on "askapatient.com", about the effectiveness and adverse
+drug events associated with Zoloft, Lexapro, Cymbalta, and Effexor XR.
+
+This dataset can be used for (multi-label) sentence classification of Adverse Drug
+Reaction (ADR), Withdrawal Symptoms (WDs), Sign/Symptoms/Illness (SSIs), Drug
+Indications (DIs), Drug Effectiveness (EF), Drug Infectiveness (INF) and Others, as well
+as for recognition of 5 different types of named entity (in the categories ADRs, WDs,
+SSIs and DIs)
+"""
+
+_HOMEPAGE = "https://www.askapatient.com/research/pharmacovigilance/corpus-ades-psychiatric-medications.asp"
+
+_LICENSE = "CC BY 4.0"
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.TEXT_CLASSIFICATION]
+
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+
+class PsyTARDataset(datasets.GeneratorBasedBuilder):
+    """The PsyTAR dataset contains patient's reviews on the effectiveness and adverse
+    drug events associated with Zoloft, Lexapro, Cymbalta, and Effexor XR."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="psytar_source",
+            version=SOURCE_VERSION,
+            description="PsyTAR source schema",
+            schema="source",
+            subset_id="psytar",
+        ),
+        BigBioConfig(
+            name="psytar_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description="PsyTAR BigBio KB schema",
+            schema="bigbio_kb",
+            subset_id="psytar",
+        ),
+        BigBioConfig(
+            name="psytar_bigbio_text",
+            version=BIGBIO_VERSION,
+            description="PsyTAR BigBio text classification schema",
+            schema="bigbio_text",
+            subset_id="psytar",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "psytar_source"
+    for config in BUILDER_CONFIGS:
+        if config.name == DEFAULT_CONFIG_NAME:
+            kwargs = asdict(config)
+
+    BUILDER_CONFIG_CLASS = partial(BigBioConfig, **kwargs)
```

Comment:
I"m starting to wonder if its worth editing every local dataset or if we just want to always make `name` a required argument. (e.g. https://github.com/bigscience-workshop/biomedical/pull/454). If we are going to edit each dataset loader, I think I prefer the custom derived class, 
* https://github.com/bigscience-workshop/biomedical/blob/master/biodatasets/psytar/psytar.py#L113
* https://github.com/bigscience-workshop/biomedical/blob/master/biodatasets/psytar/psytar.py#L147